src/booleans.js

```diff
@@ -1,62 +1,100 @@
 function negate(a) {
   // your code here
-};
+  return !a;
+}
 
 function both(a, b) {
   // your code here
-};
+  return a && b;
+}
 
 function either(a, b) {
   // your code here
-};
+  return a || b;
+}
 
 function none(a, b) {
   // your code here
-};
+  return !a && !b;
+}
 
 function one(a, b) {
   // your code here
-};
+  return (a && !b) || (!a && b);
+}
 
 function truthiness(a) {
   // your code here
-};
+  if (a === 0 || Number.isNaN(a) || a === null || a === undefined || a === '') {
+    return false;
+  }
+
+  return true;
+}
 
 function isEqual(a, b) {
   // your code here
-};
+  return a === b;
+}
 
 function isGreaterThan(a, b) {
   // your code here
-};
+  return a > b;
+}
 
 function isLessThanOrEqualTo(a, b) {
   // your code here
-};
+  return a <= b;
+}
 
 function isOdd(a) {
   // your code here
-};
+  return a % 2 !== 0;
+}
 
 function isEven(a) {
   // your code here
-};
+  return a % 2 === 0;
+}
 
 function isSquare(a) {
   // your code here
-};
+  return Math.sqrt(a) % 1 === 0;
+}
 
 function startsWith(char, string) {
   // your code here
-};
+  return char === string.charAt(0);
+}
 
 function containsVowels(string) {
   // your code here
-};
+  const vowels = ['a', 'e', 'i', 'o', 'u'];
+  let vowelCount = 0;
+  for (let i = 0; i < string.length; i += 1) {
+    for (let j = 0; j < vowels.length; j += 1) {
+      if (string[i].toLowerCase() === vowels[j]) {
+        vowelCount += 1;
+      }
+    }
+  }
+  if (vowelCount > 0) {
+    return true;
+  }
+  return false;
+}
```

Comment:
Interesting solution! While it works, the root of what the function is trying to do here is pattern recognition of the strings, in which case a regex could be a good option (as with the remove element from array solution you did). This would mean you dont have to run two for loops inside each other and can simply check if the string has any elements that match what you provide, in this case, upper and lower case vowels:
`return /[aeiou]/i.test(string);`
The characters in the brackets are what is being checked for, and the `/i` at the end stands for case-insensitive so thats also accounted for

I wouldnt expect you to have known that off hand this early on though so don't worry! Regex is one of those things that will keep coming up and it gets easier with time